### PR TITLE
Feature/image storage

### PR DESCRIPTION
**Pull request by Yassyass181104**
This pull request includes several changes to the Android test and main codebase, focusing primarily on adding new test cases for image handling and improving the UI components for image dialogs. The most important changes are grouped into test additions

### Test Additions:

* [`app/src/androidTest/java/com/android/sample/ui/activity/CreateActivityScreenTest.kt`](diffhunk://#diff-afbcad4af55e65a0899a5875391d0a7dee76c825350b09f11982f44c33ca755cL75-R109): Added tests for creating activities with images from the camera and gallery.
* [`app/src/androidTest/java/com/android/sample/ui/profile/EditProfileTest.kt`](diffhunk://#diff-fd25ce09bdafa1887e86b4f0620b8e3e208ddf1426958a306172014b0aeb1365R5-R10): Added tests for editing profiles with images from the camera and gallery. [[1]](diffhunk://#diff-fd25ce09bdafa1887e86b4f0620b8e3e208ddf1426958a306172014b0aeb1365R5-R10) [[2]](diffhunk://#diff-fd25ce09bdafa1887e86b4f0620b8e3e208ddf1426958a306172014b0aeb1365R56-R91)
* [`app/src/androidTest/java/com/android/sample/ui/profile/ProfileCreationTest.kt`](diffhunk://#diff-d3dd2195ea4e3747a8f7e6ac820b2b5f59808e6218197f03ed29f5dab91df463R62-R95): Added tests for profile creation with images from the camera and gallery.
* [`app/src/test/java/com/android/sample/model/activities/ActivitiesRepositoryFirestoreTest.kt`](diffhunk://#diff-793f5205887ff6bdef25c8978ca4f03413ee5ab271d836d0f0830049186e84b4R155-R185): Added tests for Firestore operations to ensure success and failure callbacks are called appropriately.
* [`app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt`](diffhunk://#diff-d3437fa5075f215f62585f16e0dab912f8d548129580d9aad68f258e10a86b79L4): Added multiple tests for Firestore operations related to profile and activity handling, including success and failure scenarios. [[1]](diffhunk://#diff-d3437fa5075f215f62585f16e0dab912f8d548129580d9aad68f258e10a86b79L4) [[2]](diffhunk://#diff-d3437fa5075f215f62585f16e0dab912f8d548129580d9aad68f258e10a86b79R17) [[3]](diffhunk://#diff-d3437fa5075f215f62585f16e0dab912f8d548129580d9aad68f258e10a86b79L33-L34) [[4]](diffhunk://#diff-d3437fa5075f215f62585f16e0dab912f8d548129580d9aad68f258e10a86b79L56) [[5]](diffhunk://#diff-d3437fa5075f215f62585f16e0dab912f8d548129580d9aad68f258e10a86b79L65-R170)

### CODE REVIEW

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 1]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+    `when`(mockDocumentReference.update(eq("activities"), any())).thenReturn(Tasks.forResult(null))
 
-    profileRepositoryFirestore.addProfileToDatabase(testUser, onSuccess = {}, onFailure = {})
+    profileRepositoryFirestore.addActivity("1", "activityId", onSuccess = {}, onFailure = {})
 
-    verify(mockDocumentReference).set(testUser)
+    verify(mockDocumentReference).update(eq("activities"), any())
+  }
+
+  @Test
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:80]: Verifies that update is called on Firestore when adding a liked activity.
Uses verify to check the method interaction with Firestore.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 2]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+  fun addLikedActivity_shouldCallFirestoreToAddLikedActivity() {
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forResult(null))
+
+    profileRepositoryFirestore.addLikedActivity("1", "activityId", onSuccess = {}, onFailure = {})
+
+    verify(mockDocumentReference).update(eq("likedActivities"), any())
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:89]: Verifies that update is called on Firestore when removing a liked activity.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 3]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+  fun removeLikedActivity_shouldCallFirestoreToRemoveLikedActivity() {
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forResult(null))
+
+    profileRepositoryFirestore.removeLikedActivity(
+        "1", "activityId", onSuccess = {}, onFailure = {})
+
+    verify(mockDocumentReference).update(eq("likedActivities"), any())
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:100]: good! It simulates a failure scenario for get on Firestore.
Asserts that the onFailure callback is invoked with the correct exception.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 4]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+  fun getUser_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.get()).thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.getUser(
+        "1",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:111]: Tests failure in adding an activity.
Ensures the onFailure callback is triggered with the exception.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 5]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+    `when`(mockDocumentReference.update(eq("activities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.addActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:124]: Simulates a failure in updating likedActivities.
Verifies that the failure callback receives the expected exception.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 6]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.addLikedActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:137]: Handles failures in removing a liked activity.
Confirms the failure callback execution.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 7]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+    `when`(mockDocumentReference.update(eq("likedActivities"), any()))
+        .thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.removeLikedActivity(
+        "1",
+        "activityId",
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:150]: Simulates a failure in updating a user profile.
Ensures the onFailure callback is called with the provided exception.

File: app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt [hunk 8]
```diff
@@ -62,22 +59,114 @@ class ProfilesRepositoryFirestoreTest {
...
+  fun updateProfile_shouldCallFailureCallbackOnError() {
+    val exception = Exception("Test exception")
+    `when`(mockDocumentReference.set(testUser)).thenReturn(Tasks.forException(exception))
+
+    profileRepositoryFirestore.updateProfile(
+        testUser,
+        onSuccess = { fail("Success callback should not be called") },
+        onFailure = { assert(it == exception) })
+  }
+
```
Comment (DAFERMOH) [inline app/src/test/java/com/android/sample/model/profile/ProfilesRepositoryFirestoreTest.kt:161]: Tests failures during adding a user profile to the database.
Validates that the failure callback handles the exception.

**Review by DAFERMOH — APPROVED** [review]
**Strengths**
Comprehensive test coverage for both success and failure scenarios.
Effective mocking with Mockito ensures predictable and repeatable results.
Clear test naming and readable assertions improve maintainability.
Failure cases are properly validated with appropriate exception checks.
Areas for Improvement
Add assertions to validate arguments passed in success cases.
Test edge cases (e.g., null IDs, large payloads, or network failures).
Replace hardcoded strings with constants for consistency.
Extract repetitive setups into helper methods for cleaner tests.
**Summary**
This PR provides excellent coverage and ensures robust error handling. Addressing minor improvements like edge case testing and reducing repetition will further enhance quality. Great work!
